File: Java/ex6.java
// Classe Motor
class Motor {
    private String tipo; 
    private int potencia; 

    public Motor(String tipo, int potencia) {
        this.tipo = tipo;
        this.potencia = potencia;
    }

    public String getTipo() {
        return tipo;
    }

    public int getPotencia() {
        return potencia;
    }

    public void exibirDetalhes() {
        System.out.println("Motor tipo: " + tipo + ", Potência: " + potencia + " CV");
    }
}


class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor; 

    public Carro(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor; 
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
        motor.exibirDetalhes(); 
    }

    public static void main(String[] args) {
        
        Motor motor1 = new Motor("Gasolina", 120);

        
        Carro carro1 = new Carro("Toyota", "Corolla", 2020, motor1);

      
        carro1.exibirDetalhes();
    }
}

File: Golang/ex6.go
package main

import (
	"fmt"
)

type Motor struct {
	Potencia int    // Potência do motor em cavalos de potência (CV)
	Tipo     string // Tipo de motor (ex: "Gasolina", "Álcool", "Elétrico")
}

type Carro struct {
	Marca  string
	Modelo string
	Ano    int
	Motor  Motor // Associando um objeto Motor à classe Carro
}

func main() {
	motor1 := Motor{Potencia: 150, Tipo: "Gasolina"}
	carro1 := Carro{Marca: "Toyota", Modelo: "Corolla", Ano: 2020, Motor: motor1}

	motor2 := Motor{Potencia: 200, Tipo: "Elétrico"}
	carro2 := Carro{Marca: "Tesla", Modelo: "Model 3", Ano: 2021, Motor: motor2}

	fmt.Printf("Carro: %s %s (%d) - Motor: %d CV, Tipo: %s\n", carro1.Marca, carro1.Modelo, carro1.Ano, carro1.Motor.Potencia, carro1.Motor.Tipo)
	fmt.Printf("Carro: %s %s (%d) - Motor: %d CV, Tipo: %s\n", carro2.Marca, carro2.Modelo, carro2.Ano, carro2.Motor.Potencia, carro2.Motor.Tipo)
}
